### PR TITLE
50 spring boot migration

### LINKED ISSUE #50: Migrate to Spring Boot
As a developer, I want to migrate the project configuration to Spring Boot.

This will allow easier configuration w.r.t. virtualized environments like Docker.

### PR DESCRIPTION
**Pull request by ledsoft**
Resolves #50 

It is possible to build the project as either JAR or WAR based on a Maven profile, so our deployments should require just changing the `config.properties` files to `application.yml`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ Both projects are built separately and can run separately.
 
 This section briefly lists the main technologies and principles used (or planned to be used) in the application.
 
-- Spring framework 5, Spring Security, Spring Data (paging, filtering)
+- Spring Boot, Spring Framework 5, Spring Security, Spring Data (paging, filtering)
```
Comment (psiotwo) [inline README.md]: Spring Boot 2

File: src/main/java/cz/cvut/kbss/termit/service/SystemInitializer.java [hunk 2]
```diff
@@ -38,6 +40,8 @@
 import java.util.Collections;
 import java.util.Random;
 
+@Service
+@Profile("!test")
 public class SystemInitializer {
```
Comment (psiotwo) [inline src/main/java/cz/cvut/kbss/termit/service/SystemInitializer.java:45]: where is this injected?
ServiceConfig does not seem to refer to it any more

File: src/test/java/cz/cvut/kbss/termit/rest/AdminControllerTest.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
 
     @BeforeEach
     void setUp() {
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (psiotwo) [inline src/test/java/cz/cvut/kbss/termit/rest/AdminControllerTest.java]: should be counterparted with close 
https://www.javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/MockitoAnnotations.html

File: src/test/java/cz/cvut/kbss/termit/rest/AssetControllerTest.java [hunk 4]
```diff
@@ -53,7 +53,7 @@
 
     @BeforeEach
     void setUp() {
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (psiotwo) [inline src/test/java/cz/cvut/kbss/termit/rest/AssetControllerTest.java]: dtto

File: src/test/java/cz/cvut/kbss/termit/rest/CommentControllerTest.java [hunk 5]
```diff
@@ -45,7 +45,7 @@
 
     @BeforeEach
     void setUp() {
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (psiotwo) [inline src/test/java/cz/cvut/kbss/termit/rest/CommentControllerTest.java]: dtto

File: src/test/java/cz/cvut/kbss/termit/rest/ConfigurationControllerTest.java [hunk 6]
```diff
@@ -25,7 +25,7 @@
 
     @BeforeEach
     void setUp() {
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (psiotwo) [inline src/test/java/cz/cvut/kbss/termit/rest/ConfigurationControllerTest.java]: dtto

File: src/test/java/cz/cvut/kbss/termit/rest/DataControllerTest.java [hunk 7]
```diff
@@ -58,7 +58,7 @@
 
     @BeforeEach
     void setUp() {
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (psiotwo) [inline src/test/java/cz/cvut/kbss/termit/rest/DataControllerTest.java]: dtto

File: src/test/java/cz/cvut/kbss/termit/rest/FreeRegistrationControllerTest.java [hunk 8]
```diff
@@ -41,7 +41,7 @@
 
     @BeforeEach
     void setUp() {
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (psiotwo) [inline src/test/java/cz/cvut/kbss/termit/rest/FreeRegistrationControllerTest.java]: dtto

File: src/test/java/cz/cvut/kbss/termit/rest/IdentifierControllerTest.java [hunk 9]
```diff
@@ -54,7 +54,7 @@
 
     @BeforeEach
     void setUp() {
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (psiotwo) [inline src/test/java/cz/cvut/kbss/termit/rest/IdentifierControllerTest.java]: dtto

File: src/test/java/cz/cvut/kbss/termit/rest/LanguageControllerTest.java [hunk 10]
```diff
@@ -51,7 +51,7 @@
 
     @BeforeEach
     void setUp() {
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (psiotwo) [inline src/test/java/cz/cvut/kbss/termit/rest/LanguageControllerTest.java]: dtto

File: src/test/java/cz/cvut/kbss/termit/rest/ResourceControllerTest.java [hunk 11]
```diff
@@ -92,7 +92,7 @@
 
     @BeforeEach
     void setUp() {
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (psiotwo) [inline src/test/java/cz/cvut/kbss/termit/rest/ResourceControllerTest.java]: und so weiter

**Review by psiotwo — COMMENTED** [review]
see comments

File: src/main/java/cz/cvut/kbss/termit/service/SystemInitializer.java (hunk 2 above)
Comment (ledsoft) [reply src/main/java/cz/cvut/kbss/termit/service/SystemInitializer.java:45]: It is not injected. The original factory method has been replaced by the class-level `@Service` annotation.

File: src/test/java/cz/cvut/kbss/termit/rest/AdminControllerTest.java (hunk 3 above)
Comment (ledsoft) [reply src/test/java/cz/cvut/kbss/termit/rest/AdminControllerTest.java]: Not necessarily, the documentation states that the mocks have to be closed if static methods are mocked. But I'll try to replace these with the JUnit Mockito extensions.

File: src/main/java/cz/cvut/kbss/termit/service/SystemInitializer.java (hunk 2 above)
Comment (psiotwo) [reply src/main/java/cz/cvut/kbss/termit/service/SystemInitializer.java:45]: I see, I missed the @Service annotation

**Review by psiotwo — DISMISSED** [review]

**Review by psiotwo — COMMENTED** [review]
lgtm, good idea

**Review by psiotwo — APPROVED** [review]

**Pull request merged**